src/classes/frDonation.cls

```diff
@@ -53,84 +54,136 @@ public class frDonation extends frModel implements frSyncable {
         return mappings;
     }
     
-    private Opportunity o;
+    private SObject o;
     
     public frDonation(Sync_Attempt__c syncRecord){
         super(syncRecord);
     }
-    
+
     public Boolean sync() {
+        //create Sobject
         Boolean result = false;
         Map<String, Object> request = getRequestBody();
-        o = new Opportunity();
+        for (String key : request.keySet()) {
+            System.debug('Key: ' + key + ', Value: ' + request.get(key));
+        }
+        String frId = getFunraiseId();
+
+        if(!frUtil.hasNPCobjects()) {
+            o = new Opportunity();
+            applyMappings(o, request);
+            setOpportunityStage((Opportunity) o, String.valueOf(request.get('status')));
+        } 
+        else {
+            List<SObject> gtList = Database.query('SELECT Id, Name, DonorId, AcknowledgementDate FROM GiftTransaction WHERE fr_ID__c = :frId');
+            if(gtList.isEmpty()) {
+                o = Schema.getGlobalDescribe().get('gifttransaction').newSObject();
+            } 
+            else {
+                o = gtList.get(0);
+            }
+            o.put('Name', String.valueOf(request.get('name')));
+            o.put('OriginalAmount', (Decimal) request.get('amount'));
+            o.put('PaymentMethod', String.valueOf(request.get('paymentMethodType')));
+            o.put('AcknowledgementDate', System.today());
```

Comment:
I think using the System.today() value here is incorrect as Funraise user has the ability to resync a transaction, or if an update is made to the transaction it will sync to Salesforce which will hit this code path.  That will end up changing the AcknowledgementDate to whenever the most recent sync was, rather than the original date of the gift

Reply:
AcknowledgementDate is a required field for gift transaction so we need to ensure this date is populated with something, is there a donation request field you could suggest to populate this?

Reply:
Ok got it, so the way we have handled that for opportunity and contact is that the default mappings we provide handle setting all of the required fields (Stage, CloseDate, Name on Opportunity).  So I think this should be replaced with suggested default mappings in frSetupController

Reply:
Removed this and added defaults for the objects!